src/west/log.py

```diff
@@ -166,20 +166,25 @@ def msg(*args, color=None, stream=sys.stdout):
     if _use_colors():
         _reset_colors(stream)
 
+def use_color():
+    '''Returns True if the configuration requests colored output.'''
+    return _use_colors(warn=False)
+
 _COLOR_UI_WARNED = False
 
-def _use_colors():
+def _use_colors(warn=True):
     # Convenience function for reading the color.ui setting
     try:
         return config.config.getboolean('color', 'ui', fallback=True)
     except ValueError as e:
-        global _COLOR_UI_WARNED
-        if not _COLOR_UI_WARNED:
-            print(f"WARNING: invalid color.ui value: {e}.",
-                  file=sys.stderr)
-            print('         To fix, run: "west config color.ui <true|false>"',
-                  file=sys.stderr)
-            _COLOR_UI_WARNED = True
+        if warn:
+            global _COLOR_UI_WARNED
+            if not _COLOR_UI_WARNED:
+                print(f"WARNING: invalid color.ui value: {e}.",
+                      file=sys.stderr)
+                print('         To fix: "west config color.ui <true|false>"',
```

Comment:
didn't knew `west` had this feature, and the changes themselves looks good to me.

But as we are using: `config.ui`, and git is also having a `config.ui` setting, then maybe we should align allowed values.

That is, `auto`, `always`, `false`

From https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration
> color.ui
>
> Git automatically colors most of its output, but there’s a master switch if you don’t like this behavior. To turn off all Git’s colored terminal output, do this:
>
> `$ git config --global color.ui false`
>
> The default setting is auto, which colors output when it’s going straight to a terminal, but omits the color-control codes when the output is redirected to a pipe or a file.
>
> You can also set it to always to ignore the difference between terminals and pipes. You’ll rarely want this; in most scenarios, if you want color codes in your redirected output, you can instead pass a --color flag to the Git command to force it to use color codes. The default setting is almost always what you’ll want.

Reply:
While I don't disagree with the suggestion, that feels like scope creep.

Reply:
> While I don't disagree with the suggestion, that feels like scope creep.

That was not my intention.
I was more thinking that it would be just passing on the setting, as well as not reinventing the wheel.

But it is in principle not something which needs discussion in this PR.

Out of curiosity, why is this needed as a west config also, when user can already do:
```
git config --global color.ui auto
```
and we could just honor that setting ?

Reply:
The west option has the same name as the git one for familiarity, but it controls west's color output, e.g. for log.wrn().

Reply:
... and therefore subprocesses which are capable of colored output (like git) should be encouraged to print in color when the west user has requested west colors.

Reply:
Thanks.